paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/GooglePayButton.kt

```diff
@@ -46,23 +69,99 @@ internal class GooglePayButton @JvmOverloads constructor(
         // Call super so we don't inadvertently effect the primary button as well.
         super.setClickable(true)
         super.setEnabled(true)
-        viewBinding.googlePayPrimaryButton.backgroundTintList = null
-        viewBinding.googlePayPrimaryButton.finishedBackgroundColor = Color.TRANSPARENT
+    }
+
+    fun initialize(
+        cornerRadius: Int,
+        allowCreditCards: Boolean,
+        billingAddressParameters: GooglePayJsonFactory.BillingAddressParameters?
+    ) {
+        initializePrimaryButton()
+        val isDark = context.isSystemDarkTheme()
+        val allowedPaymentMethods = GooglePayJsonFactory(context)
+            .createIsReadyToPayRequest(
+                billingAddressParameters = billingAddressParameters,
+                allowCreditCards = allowCreditCards
+            ).getJSONArray(ALLOWED_PAYMENT_METHODS).toString()
+
+        if (GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(context) != 0) return
+
+        viewBinding.googlePayPaymentButton.initialize(
+            ButtonOptions.newBuilder()
+                .setButtonTheme(
+                    if (isDark) {
+                        ButtonConstants.ButtonTheme.LIGHT
+                    } else {
+                        ButtonConstants.ButtonTheme.DARK
+                    }
+                )
+                .setButtonType(ButtonConstants.ButtonType.PAY)
+                .setCornerRadius(cornerRadius)
+                .setAllowedPaymentMethods(allowedPaymentMethods)
+                .build()
+        )
+    }
+
+    private fun initializePrimaryButton() {
+        with(viewBinding.googlePayPrimaryButton) {
+            val isDark = context.isSystemDarkTheme()
+            setAppearanceConfiguration(
+                StripeTheme.primaryButtonStyle,
+                null
+            )
+            if (isDark) {
+                val backgroundColor = ContextCompat.getColor(
+                    context,
+                    R.color.stripe_googlepay_color
+                )
+                finishedBackgroundColor = backgroundColor
+                backgroundTintList = ColorStateList.valueOf(backgroundColor)
+                setLockIconDrawable(
+                    R.drawable.stripe_ic_lock_googlepay_dark
```

Comment:
Instead of using different drawables based on the theme, can we just have one version of the drawable that internally uses different _colors_ based on the theme?